### PR TITLE
front: correctly type ChartHelpers

### LINKED ISSUE #3210: Add missing Types for SimulationResults/ChartHelpers/ChartHelpers.ts
correctly type ChartHelpers.ts

This issue follows this PR comment :
https://github.com/DGEXSolutions/osrd/pull/3012#discussion_r1088809056

### PR DESCRIPTION
**Pull request by anisometropie**
closes #3210

### CODE REVIEW

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts [hunk 1]
```diff
@@ -234,46 +229,45 @@ export const mergeDatasArea = <T>(
   return [];
 };
 
-export const mergeDatasAreaConstant = (
-  data1: Array<any>,
-  data2: Array<any>,
-  keyValues: Array<any>
+// called with keyValues
+// ['position', 'gradient']
+// ['position', 'speed']
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts:234]: Why not use an enum here to specify keyValues ?

Comment (anisometropie) [reply front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/ChartHelpers.ts:234]: This is just help to know what the function is called with. it’s hard to follow the chain of calls.
Not sure how to use an enum here.

File: front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/__tests__/ChartHelpers.spec.ts [hunk 2]
```diff
@@ -1,7 +1,64 @@
-import { mergeDatasArea } from '../ChartHelpers';
+import { Regime, SimulationTrain } from 'reducers/osrdsimulation/types';
+import { mergeDatasArea, interpolateOnTime } from '../ChartHelpers';
+import { LIST_VALUES_NAME_SPACE_TIME } from '../../simulationResultsConsts';
+
+import train from '../../../../../../../tests/assets/operationStudies/train.json';
+import simulationTrain from '../../../../../../../tests/assets/operationStudies/simulationTrain';
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/__tests__/ChartHelpers.spec.ts:6]: Can't use absolute paths ?

Comment (anisometropie) [reply front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/__tests__/ChartHelpers.spec.ts:6]: Not here, it’s not in the `src` directory. I’ll try to do an alias when we have more cases like this, but it needs more config.

**Review by nicolaswurtz — APPROVED** [review]
